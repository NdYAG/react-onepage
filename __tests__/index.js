import { Component } from 'react'
import renderer from 'react-test-renderer'
import Slider from '../src'

const events = {}
document.addEventListener = jest.fn((event, callback) => {
  events[event] = callback
})

class App extends Component {
  constructor() {
    super()
    this.state = { pageIndex: 0 }
  }
  handlePage(pageIndex) {
    this.setState({ pageIndex })
  }
  render() {
    return (
      <Slider
        pageIndex={this.state.pageIndex}
        onPage={this.handlePage.bind(this)}>
        <div>0</div>
        <div>1</div>
        <div>2</div>
      </Slider>
    )
  }
}

const slider = <App />

it('renders correctly', () => {
  const tree = renderer.create(slider).toJSON()
  expect(tree).toMatchSnapshot()
})

it('slides to prev/next when hit arrow up/down', done => {
  const component = renderer.create(slider)
  let tree
  events.keydown({ key: 'ArrowDown' })
  setTimeout(() => {
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    events.keydown({ key: 'ArrowUp' })
    setTimeout(() => {
      tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      done()
    }, 2000)
  }, 2000)
})

it('slides to prev/next when mouse wheel forwards/backwards', done => {
  const component = renderer.create(slider)
  let tree
  events.wheel({ deltaY: 20 })
  setTimeout(() => {
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    events.wheel({ deltaY: -20 })
    setTimeout(() => {
      tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      done()
    }, 2000)
  }, 2000)
})
