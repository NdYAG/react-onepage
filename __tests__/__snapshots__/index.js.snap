// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransform": "translate3d(0, 0px, 0)",
        "height": "100%",
        "transform": "translate3d(0, 0px, 0)",
      }
    }
  >
    <div>
      0
    </div>
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;

exports[`slides to prev/next when hit arrow up/down 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransform": "translate3d(0, -768px, 0)",
        "height": "100%",
        "transform": "translate3d(0, -768px, 0)",
      }
    }
  >
    <div>
      0
    </div>
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;

exports[`slides to prev/next when hit arrow up/down 2`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransform": "translate3d(0, 0px, 0)",
        "height": "100%",
        "transform": "translate3d(0, 0px, 0)",
      }
    }
  >
    <div>
      0
    </div>
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;

exports[`slides to prev/next when mouse wheel forwards/backwards 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransform": "translate3d(0, -768px, 0)",
        "height": "100%",
        "transform": "translate3d(0, -768px, 0)",
      }
    }
  >
    <div>
      0
    </div>
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;

exports[`slides to prev/next when mouse wheel forwards/backwards 2`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransform": "translate3d(0, 0px, 0)",
        "height": "100%",
        "transform": "translate3d(0, 0px, 0)",
      }
    }
  >
    <div>
      0
    </div>
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;
